planner/core/find_best_task.go

```diff
@@ -2377,6 +2375,22 @@ func (ds *DataSource) convertToPointGet(prop *property.PhysicalProperty, candida
 		if partitionInfo == nil {
 			return invalidTask
 		}
+	} else if pi != nil && ds.SCtx().GetSessionVars().StmtCtx.UseDynamicPruneMode {
+		// dynamic prune
+		idxs, err := PartitionPruning(ds.ctx, ds.table.GetPartitionedTable(), ds.allConds, ds.partitionNames, ds.TblCols, ds.names)
```

Comment:
Do `convertToBatchPointGet` need the same change?

Reply:
No, `BatchPointGet` will get the pid info in `BatchPointGetPlan.Init()`